### ISSUE TITLE
Use Basic Auth alongside with NTLM

### ISSUE THREAD

**Issue by guderkar** [gautamsi/ews-javascript-api#157]
Hi, I need to authenticate either with Basic Auth or NTLM, because I don't know if server will be using Basic or NTLM or both. So I just set the exch.XHRApi property.
```javascript
var ntlmXHR = require("./ntlmXHRApi");
var ntlmXHRApi = new ntlmXHR.ntlmXHRApi(username, password);
var exch = new ews.ExchangeService(ews.ExchangeVersion.Exchange2007);
exch.Credentials = new ews.ExchangeCredentials(username, password);
exch.Url = new ews.Uri(URL);
exch.XHRApi = ntlmXHRApi;

exch.FindFolders(ews.WellKnownFolderName.Root, new ews.FolderView(1)).then(function (response) {
	process.stdout.write("OK");
        process.exit(0);
}, function (error) {
	process.stdout.write("ERROR");
	process.exit(1);
});
```
When only NTML is enabled on the server it's working fine.
When NTLM + Basic Auth is enabled it's also working fine.
But when only Basic Auth is enabled I get: TypeError: Cannot set property 'getAllResponseHeaders' of undefined.
Is there some workaround for this?

**Comment by gautamsi** [comment]
Authentication has to be configured per server, means you can call with ntlm or basic auth support but not both.  there is a way to figure this out. You can call EWS with random data, it will come back with response header with what type of authentication is supported, look for `WWW-Authenticate` header, if it contains `Basic` word, you do not need to wire up any authentication. 

once i publish update to Autodiscover, you can then use correct way to determine authentication.

You may also try `ews-javascript-api@next` npm package with [`ews-javascript-api-auth`](/gautamsi/ews-javascript-api-auth) package, which simplifies authentication.

**Comment by gautamsi** [comment]
have you checked out #180  this will help your scenario. Also published in 0.9.0 final.
